### PR TITLE
Claim deed

### PR DESCRIPTION
**Pull request by Rantaa**

### CODE REVIEW

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package com.goodpeople.gooddeeds.Model;
 
+import androidx.annotation.Nullable;
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java]: what is this?

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java [hunk 2]
```diff
@@ -316,20 +319,64 @@ public void createRequest(String subject, String description) {
...
 
+    /**
+     * Method for checking that the logged in account is not
+     * the creator of the deed
+     *
+     * @return true if the creator of the deed is the logged in account
+     * false otherwise
+     */
+    public boolean isMyOwnDeed() {
+        return ((currentDeed.getGivingAccount() == loggedInAccount)
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java]: Nice solution :)

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java [hunk 3]
```diff
@@ -316,20 +319,64 @@ public void createRequest(String subject, String description) {
...
+     * Method for claiming deed. Sets, whichever is not already initialized of,
+     * givingAccount or receivingAccount to the loggedInAccount.
+     */
+    public void claimDeed() {
+        if (currentDeed.getReceivingAccount() == null) {
+            currentDeed.setReceivingAccount(loggedInAccount);
+        } else if (currentDeed.getGivingAccount() == null) {
+            currentDeed.setGivingAccount(loggedInAccount);
+        }
+    }
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java]: Nice general solutions that work for both req and offers

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java (hunk 1 above)
Comment (Rantaa) [reply goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java]: No idea, must have been imported during implementation... it's unused though, so I'll remove it.

**Comment by Rantaa** [comment]
> Seems like I can press claim many times without anything really happening?

It didn't show until the market-view was pressed again. Fixed so that it refreshes on resume.

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/Repositories/DeedRepositoryImpl.java [hunk 4]
```diff
@@ -115,4 +115,19 @@ public void setCurrentDeed(UUID id) {
     public boolean isMyActiveDeed() {
         return goodDeeds.isMyActiveDeed();
     }
+
+    @Override
+    public boolean isClaimed() {
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/Repositories/DeedRepositoryImpl.java]: should have javadoc on all here?

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Controller/DeedController.java [hunk 5]
```diff
@@ -103,4 +102,18 @@ public void setCurrentDeedHandler(UUID uuid) {
     public boolean isMyActiveDeedHandler() {
         return deedRepository.isMyActiveDeed();
     }
+
+    public boolean isClaimedHandler() {
+        return deedRepository.isClaimed();
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Controller/DeedController.java]: should have javadoc on all public here?

**Pull request merged**